EpiAware/src/EpiInfModels/Renewal.jl

```diff
@@ -111,6 +111,166 @@ function (epi_model::Renewal)(recent_incidence, Rt)
+
+# Returns
+The initial vector of infected individuals.
+
+"""
+function make_renewal_init(epi_model::Renewal, I₀, Rt)
+    r_approx = R_to_r(Rt[1], epi_model)
+    return I₀ * [exp(-r_approx * t) for t in 0:(epi_model.data.len_gen_int - 1)]
+end
+
+@doc raw"
+Model unobserved/latent infections as due to time-varying Renewal model with reproduction
+number ``\mathcal{R}_t`` which is generated by a latent process and a population size
+of available people who can be infected `N`.
+
+## Mathematical specification
+
+If ``Z_t`` is a realisation of the latent model, then the unobserved/latent infections are
+given by
+
+```math
+\begin{align}
+\mathcal{R}_t &= g(Z_t),\\
+S_t &= S_{t-1} - I_t,\\
+I_t &= {S_{t-1} \over N}\mathcal{R}_t \sum_{i=1}^{n-1} I_{t-i} g_i, \qquad t \geq 1, \\
+I_t &= g(\hat{I}_0) \exp(r(\mathcal{R}_1) t), \qquad t \leq 0.
+\end{align}
+```
+
+where ``g`` is a transformation function and the unconstrained initial infections
+``\hat{I}_0`` are sampled from a prior distribution. The discrete generation interval is
+given by ``g_i``.
+
+``r(\mathcal{R}_1)`` is the exponential growth rate implied by ``\mathcal{R}_1)``
+using the implicit relationship between the exponential growth rate and the reproduction
+number.
+
+```math
+\mathcal{R} \sum_{j \geq 1} g_j \exp(- r j)= 1.
+```
+
+`Renewal` are constructed by passing an `EpiData` object `data` and an
+`initialisation_prior` for the prior distribution of ``\hat{I}_0``. The default
+`initialisation_prior` is `Normal()`.
+
+## Constructor
+
+- `RenewalWithPopulation(; data, initialisation_prior, pop_size)`.
+
+## Example usage with `generate_latent_infs`
+
+`generate_latent_infs` can be used to construct a `Turing` model for the latent infections
+conditional on the sample path of a latent process. In this example, we generate a sample
+of a white noise latent process.
+
+First, we construct an `Renewal` struct with an `EpiData` object, an initialisation
+prior and a transformation function.
+
+```julia
+using Distributions, Turing, EpiAware
+gen_int = [0.2, 0.3, 0.5]
+g = exp
+
+# Create an EpiData object
+data = EpiData(gen_int, g)
+
+# Create an Renewal model
+renewal_model = RenewalWithPopulation(data = data, initialisation_prior = Normal(), pop_size = 1e6)
+```
+
+Then, we can use `generate_latent_infs` to construct a Turing model for the unobserved
+infection generation model set by the type of `direct_inf_model`.
+
+```julia
+# Construct a Turing model
+Z_t = randn(100) * 0.05
+latent_inf = generate_latent_infs(renewal_model, Z_t)
+```
+
+Now we can use the `Turing` PPL API to sample underlying parameters and generate the
+unobserved infections.
+
+```julia
+# Sample from the unobserved infections model
+
+#Sample random parameters from prior
+θ = rand(latent_inf)
+#Get unobserved infections as a generated quantities from the model
+I_t = generated_quantities(latent_inf, θ)
+```
+"
+@kwdef struct RenewalWithPopulation{S <: Sampleable} <: EpiAwareBase.AbstractTuringRenewal
```

Comment:
My current thinking is we want one renewal struct that then has a field that takes a struct which dictates the scan function (and maybe the initialisation model?) We can then compose these all together? 

Alternatively we rethink this whole thing (the inf module) as the composition of some later seeding and scan process.